src/index.js

```diff
@@ -1,39 +1,72 @@
 import './style.css';
 import init from './init.js';
 import checkBoxEvent from './checkBoxEvent.js';
+import addTodo from './addTodo.js';
+import clearAllCompleted from './clearAllCompleted.js';
+import deleteTodo from './deleteTodo.js';
+import updateTodo from './updateTodo.js';
 
 const todos = document.getElementById('todo-container');
+const inputField = document.getElementById('input-field');
 
 class TodoList {
   constructor() {
     this.list = [];
   }
 
   static displayTodo = () => {
-    init(this.list);
-    this.list = [...init(this.list)];
+    this.list = init(this.list);
     let htmlCode = '';
     this.list.forEach((item) => {
       htmlCode += `
       <div class='todo-item'>
         <div class='left-items'>
           <input id='boxes' type="checkbox" ${item.completed ? 'checked' : ''} data-index=${item.index}>
-          <p>${item.description}</p>
+          <p class='description' contenteditable=${!item.completed} data-index=${item.index}>${item.description}</p>
         </div>
-        <button class='remove-btn'>del</button>
+        <button class='remove-btn'data-index=${item.index}>del</button>
       </div>
       `;
     });
-    todos.insertAdjacentHTML('beforeend', htmlCode);
+    todos.innerHTML = htmlCode;
 
     document.querySelectorAll('#boxes').forEach((item) => {
       item.addEventListener('change', () => {
         checkBoxEvent(item);
       });
     });
+
+    document.querySelectorAll('.remove-btn').forEach((item) => {
+      item.addEventListener('click', (e) => {
+        deleteTodo(e);
+        this.displayTodo();
+      });
+    });
+
+    document.querySelectorAll('.description').forEach((item) => {
+      item.addEventListener('keypress', (e) => {
+        updateTodo(e);
+      });
```

Comment:
Look at the image below
- [ ] When I try to edit a to-do task and make only one change it does not reflect on the localStorage and so does not appear on reload. I have to make another change for my **previous** update to appear. Here, I would suggest changing **`'keypress'`** to **`'input'`** so that your update action works while the edit is being made in real-time.

![nas-edit](https://user-images.githubusercontent.com/36542986/144484236-d57f8ba6-bbed-466a-8710-1ac30e4bd389.jpg)

Reply:
okay @TedLivist thanks for the feedbacks